driftbase/flexmatch.py

```diff
@@ -880,10 +919,13 @@ def _post_value_loaded(self):
             last_ban_date = self._value.get("last_ban_date")
             if isinstance(last_ban_date, str):
                 self._value["last_ban_date"] = datetime.fromisoformat(last_ban_date)
+            custom_unban_date = self._value.get("custom_unban_date")
+            if isinstance(custom_unban_date, str):
+                self._value["custom_unban_date"] = datetime.fromisoformat(custom_unban_date)
 
     def __exit__(self, exc_type, exc_val, exc_tb):
-        if self._lock.owned():
-            with g.redis.conn.pipeline() as pipe:
+        if self._lock is not None and self._lock.owned():
+            with self._redis.conn.pipeline() as pipe:
```

Comment:
Would you need exception handling here to make sure the lock is always released even if the delete/set throws?